app/assets/javascripts/discourse/app/lib/plugin-api.gjs

```diff
@@ -3379,6 +3380,10 @@ class PluginApi {
     registeredTabs.push(tab);
   }
 
+  registerRichEditorExtension(extension) {
```

Comment:
Would be good to add a minimal description here, even if it's just "EXPERIMENTAL: This API will change without warning" for now